sql/lexer.go

```diff
@@ -138,8 +138,7 @@ func (l *lexer) emitIdentOrKeyword(lval *sqlSymType) int {
 }
 
 var (
-	// Stolen https://www.regular-expressions.info/floatingpoint.html.
-	reNumber = regexp.MustCompile("[-+]?[0-9]*.?[0-9]+([eE][-+]?[0-9]+)?")
+	reNumber = regexp.MustCompile("[-+]?[0-9]*[.]?[0-9]+([eE][-+]?[0-9]+)?")
```

Comment:
`.` will match any character. What we want is the exact match of character `.`.

Reply:
does that mean should `\.` also works?